simulator/src/scenarios/common.rs

```diff
@@ -0,0 +1,204 @@
+use std::collections::BTreeSet;
+
+use dryoc::keypair::StackKeyPair;
+use dsnp_graph_config::{DsnpUserId, Environment, GraphKeyType, SchemaId};
+use dsnp_graph_core::api::{
+	api::{GraphAPI, GraphState},
+	api_types::{Action, Connection, GraphKeyPair, ImportBundle},
+};
+use rand::{prelude::SliceRandom, thread_rng, Rng};
+use rayon::iter::{IntoParallelRefIterator, ParallelIterator};
+
+use crate::GlobalState;
+
+pub fn add_keys_for_users(
+	env: Environment,
+	state: &mut GlobalState,
+	selected_users: &Vec<DsnpUserId>,
+	schema_id: SchemaId,
+) {
+	let keys_updates: Vec<_> = selected_users
+		.par_iter()
+		.map(|user_id| {
+			let key_pair_raw = StackKeyPair::gen();
+			let graph_key_pair = GraphKeyPair {
+				secret_key: key_pair_raw.secret_key.to_vec(),
+				public_key: key_pair_raw.public_key.to_vec(),
+				key_type: GraphKeyType::X25519,
+			};
+
+			let (existing_keys, key_pairs, pages, _) =
+				state.get_all_data_for_user(env.clone(), *user_id, schema_id);
+			let mut graph = GraphState::new(env.clone());
+			graph
+				.import_users_data(&vec![ImportBundle {
+					dsnp_keys: Some(existing_keys.clone()),
+					dsnp_user_id: *user_id,
+					schema_id,
+					key_pairs,
+					pages,
+				}])
+				.expect("Should import data");
+			graph
+				.apply_actions(
+					&vec![Action::AddGraphKey {
+						owner_dsnp_user_id: *user_id,
+						new_public_key: graph_key_pair.public_key.clone(),
+					}],
+					&None,
+				)
+				.expect("error adding graph key");
+
+			let updates = graph.export_updates().expect("error exporting updates");
+
+			println!("importing keys for user {}", user_id);
```

Comment:
This is REALLY noisy... (also in one other place, same message). Maybe we don't need this debug line?

Reply:
Hmm, not so noisy now that I am running with the right version of the SDK built 😆